### PR TITLE
feat(sweep): Add tool to extract best hyperparameters from WandB sweeps

### PR DESCRIPTION
**Pull request by akrbc9**
### **Summary**
This PR adds a new tool `sweep_best_params.py` that extracts the best performing hyperparameters from completed WandB sweeps and generates Hydra-compatible configuration patches for easy reuse.

#### **New Tool: `sweep_best_params.py`**
- **Extracts best hyperparameters** from completed WandB sweeps
- **Generates Hydra-compatible config patches** with `@package _global_` directive
- **Supports multiple output formats**: YAML patches, CLI overrides, complete commands

#### **📊 Analysis Options**
- **`--top-n`**: Analyze top N runs (default: 1)
- **`--show-scores`**: Display scores for all successful runs
- **`--no-patch`**: Skip patch file generation
- **`--output-dir`**: Custom output directory for patches

#### **📚 Documentation**
- **Updated sweep README** with comprehensive usage instructions
- **Added example configurations** and usage patterns
- **Included troubleshooting section** for common issues

#### **🧪 Testing**
- **Comprehensive test suite** with 303 lines of test coverage
- **Tests multiple output formats** and edge cases
- **Validates Hydra compatibility** of generated patches

### **Usage Example**
```bash
# Extract best hyperparameters from a sweep
python tools/sweep_best_params.py +sweep_name=<sweep_name>

# Analyze top 3 runs and show all scores
python tools/sweep_best_params.py +sweep_name=<sweep_name> --top-n 3 --show-scores

```

### **Output Format**
1. **YAML Patch**: Hydra-compatible configuration patch file

### **Impact**
This tool significantly improves the sweep workflow by:
- **Automating hyperparameter extraction** from successful sweeps
- **Enabling easy reuse** of best configurations
---

**Files Added:** 4 files, 760 lines
- `tools/sweep_best_params.py` (346 lines)
- `tests/tools/test_sweep_best_params.py` (303 lines)
- `metta/sweep/README.md` (104 lines)
- `configs/trainer/patch/axel_remote_test_1_best.yaml` (7 lines)

[Asana Task](https://app.asana.com/1/1209016784099267/project/1210348820405981/task/1210747172549388)

### CODE REVIEW

**Comment by akrbc9** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Metta-AI/metta/1375?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1375** <a href="https://app.graphite.dev/github/pr/Metta-AI/metta/1375?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/Metta-AI/metta/1375?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1287** <a href="https://app.graphite.dev/github/pr/Metta-AI/metta/1287?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: configs/sweep_job.yaml [hunk 1]
```diff
@@ -11,8 +11,8 @@ cmd: sweep
 sweep_run: null  # Will be set via command line override
 
 trainer:
-  total_timesteps: 5_000_000   # 5M timesteps for proper RL training (was 50k - too short!)
-  curriculum: /env/mettagrid/curriculum/arena
+  total_timesteps: 500_000_000   # 5M timesteps for proper RL training (was 50k - too short!)
```
[the remark this answers is not in the document]
Comment (daveey) [reply configs/sweep_job.yaml:14]: this is 500M, update the comment?

File: tools/sweep_best_params.py [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+import yaml
+from omegaconf import DictConfig, OmegaConf
+
+from metta.common.util.logging_helpers import setup_mettagrid_logger
+from metta.common.wandb.sweep import sweep_id_from_name
+
+logger = setup_mettagrid_logger("sweep_best_params")
+
+
+def convert_to_yaml(data: Union[str, Dict[str, Any]]) -> str:
```
Comment (daveey) [inline tools/sweep_best_params.py:22]: put in common/util/yaml.py?

File: tools/sweep_best_params.py [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+                data = ast.literal_eval(data)
+            except (ValueError, SyntaxError) as e:
+                logger.warning(f"Failed to parse string as dict: {e}")
+                return data
+
+    # Convert to YAML
+    return yaml.dump(data, default_flow_style=False, sort_keys=False)
+
+
+def get_sweep_runs(sweep_id: str, entity: str, project: str) -> List[Any]:
```
Comment (daveey) [inline tools/sweep_best_params.py:45]: this should go somewhere in sweep/wandb.py or something

File: tools/sweep_best_params.py [hunk 4]
```diff
@@ -0,0 +1,280 @@
+#!/usr/bin/env -S uv run
```
Comment (daveey) [inline tools/sweep_best_params.py:1]: rename file to get_best_params_from_sweep

**Review by daveey — APPROVED** [review]
